src/khoj/database/adapters/__init__.py

```diff
@@ -405,9 +405,9 @@ def get_conversation_by_user(
         else:
             conversation = (
                 Conversation.objects.filter(user=user, client=client_application).order_by("-updated_at").first()
-            )
+            ) or Conversation.objects.create(user=user, client=client_application)
 
-        return conversation or Conversation.objects.create(user=user, client=client_application)
+        return conversation
```

Comment:
The conversation returned would be `None` then if the `conversation_id` did not match any. Is that what we want to do?

Reply:
Yes, this is the core fix being used for the bug here. Also it's a bit strange if I want a get conversation with specific id but can instead get a conversation with a different id when a conversation with requested id doesn't exist